packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -340,6 +344,96 @@ export function useSelectableCollection(options: SelectableCollectionOptions): S
     }
   }, [isVirtualized, scrollRef, manager.focusedKey]);
 
+  let prevScroll = useRef({scrollTop: scrollRef?.current?.scrollTop || 0, scrollLeft: scrollRef?.current?.scrollLeft || 0});
+  // Bring focused key into view if focus enters the collection from outside. Only for non virtualized collections,
+  // virtualized collections should handle this in their virtualizer or get this for free via our Virtualizer component
+  useEffect(() => {
+    let onFocus = (e) => {
+      if ((e.target instanceof HTMLElement && ref.current?.contains(e.target)) && !isVirtualized && !manager.isFocused) {
+        if (manager.focusedKey) {
+          let element = ref.current.querySelector(`[data-key="${manager.focusedKey}"]`) as HTMLElement;
+          if (element) {
+            // Figure out if element is out of view
+            let scrollContainerTop = scrollRef.current.offsetTop + prevScroll.current.scrollTop;
+            let scrollContainerBottom = scrollRef.current.offsetTop + prevScroll.current.scrollTop + scrollRef.current.offsetHeight;
+            let scrollContainerLeft = scrollRef.current.offsetLeft + prevScroll.current.scrollLeft;
+            let scrollContainerRight = scrollRef.current.offsetLeft + prevScroll.current.scrollLeft + scrollRef.current.offsetWidth;
+            let elementTop = element.offsetTop;
+            let elementBottom = element.offsetTop + element.clientHeight;
+            let elementLeft = element.offsetLeft;
+            let elementRight = element.offsetLeft + element.clientWidth;
+
+            let elementOutOfView = elementTop <= scrollContainerTop || elementBottom >= scrollContainerBottom || elementRight >= scrollContainerRight || elementLeft <= scrollContainerLeft;
+            // If focused key is out of view and is in the scrollable region, scroll it into view when re-entering the table
+            if (scrollRef.current?.contains(element) && elementOutOfView) {
+              scrollIntoView(scrollRef.current, element);
+            } else {
+              // If focusedkey is already in view, override the scroll that may happen from shift tabbing (browser will focus last focusable element in the table which may be out of view, causing a scroll)
+              scrollRef.current.scrollTop = prevScroll.current.scrollTop;
+              scrollRef.current.scrollLeft = prevScroll.current.scrollLeft;
+            }
+          }
+        }
+      }
+    };
+
+    window.addEventListener('focus', onFocus, true);
+    return () => {
+      window.removeEventListener('focus', onFocus, true);
+    };
+  }, [manager.focusedKey, scrollRef, isVirtualized, ref]);
+
+  // Save the previous scroll position
+  let onScroll = useCallback(() => {
+    prevScroll.current = {scrollTop: scrollRef.current.scrollTop, scrollLeft: scrollRef.current.scrollLeft};
+  }, [scrollRef]);
+
+  useEffect(() => {
+    let onKeyDown = (e) => {
```

Comment:
This block was added to block the scrollable region from scrolling since shift tabbing back into the table will focus the last checkbox in the table before it shifts focus back to the tableview's focused key -> causes a scroll to happen -> messes up the calculation of scrollIntoView.

Problem is that this doesn't quite work when tabbing from outside the window (e.g. browser bar)...

Reply:
There also exists a bug for non-virtualized tables where tabbing from the browser bar to the table will focus the select all checkbox -> causes the focused key to update, overwriting the previously focused key if one existed.
https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/grid/src/useGridCell.ts#L200-L202 is supposed to stop cases like that but `isFocusVisible` doesn't return true when tabbing from the browser bar. This is because https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/useFocusVisible.ts#L132 won't capture the tab keydown from the browser bar to the table since it is outside the document